src/main/java/org/broadinstitute/hellbender/engine/AssemblyRegionWalker.java

```diff
@@ -305,7 +310,7 @@ public final void traverse() {
      * @param features FeatureManager
      */
     private void processReadShard(Shard<GATKRead> shard, ReferenceDataSource reference, FeatureManager features ) {
-        final Iterator<AssemblyRegion> assemblyRegionIter = new AssemblyRegionIterator(shard, getHeaderForReads(), reference, features, assemblyRegionEvaluator(), minAssemblyRegionSize, maxAssemblyRegionSize, assemblyRegionPadding, activeProbThreshold, maxProbPropagationDistance);
+        final Iterator<AssemblyRegion> assemblyRegionIter = new AssemblyRegionIterator(shard, getHeaderForReads(), reference, features, assemblyRegionEvaluator(), minAssemblyRegionSize, maxAssemblyRegionSize, assemblyRegionPadding, activeProbThreshold, maxProbPropagationDistance, includeReadsWithDeletionsInIsActivePileups());
```

Comment:
Perhaps you could add a test at the `AssemblyRegionIterator` that enforces that this variable is making a difference? It looks like the `LocusIteratorByState` has tests with both values set but its unclear that its behavior is enforced.

Reply:
I've added a unit test for `AssemblyRegionIterator` that proves that this feature causes reads with deletions to be retained in the pileups sent to the `AssemblyRegionEvaluator`.